### PR TITLE
Add arcade telemetry for build failures in Build-Repo

### PR DESCRIPTION
**Pull request by Forgind**
Helps us correctly classify build errors as such when running them through Arcade.

### CODE REVIEW

File: eng/build.ps1 [hunk 1]
```diff
@@ -174,9 +174,8 @@ try {
   }
 }
 catch {
-  Write-Host $_
-  Write-Host $_.Exception
   Write-Host $_.ScriptStackTrace
+  Write-PipelineTelemetryError -Category "Build" -Message $_
```
Comment (rainersigwald) [inline eng/build.ps1]: Is this file is the only relevant change? It looks like you pulled in some other commits from another PR.

Comment (Forgind) [reply eng/build.ps1]: You're right—I pulled in the parallelism branch by accident. I'll sort that out.

**Comment by rainersigwald** [comment]
The root cause of the Arcade misclassification is that we're not using the Arcade logger in our Windows legs. To fix it, we should pick up #4839. I think we should abandon this PR in favor of that one.

**Comment by rainersigwald** [comment]
Closing in favor of #4839.
